src/serializationHelpers.ts

```diff
@@ -173,3 +319,462 @@ export function serializeYearMonth(yearMonth: string): Buffer {
+            if (typeof BigInt(userInput) === 'bigint') {
+                return encodeWord128(BigInt(userInput));
+            } else {
+                throw new Error('Unsigned integer required');
+            }
+        case ParameterType.I8:
+            if (typeof userInput === 'number') {
+                return encodeInt8(userInput as number);
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.I16:
+            if (typeof userInput === 'number') {
+                return encodeInt16(userInput as number);
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.I32:
+            if (typeof userInput === 'number') {
+                return encodeInt32(userInput as number);
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.I64:
+            if (typeof BigInt(userInput) === 'bigint') {
+                return encodeInt64(BigInt(userInput));
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.I128:
+            if (typeof BigInt(userInput) === 'bigint') {
+                return encodeInt128(BigInt(userInput));
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.Bool:
+            if (typeof userInput === 'boolean') {
+                return encodeBool(userInput as boolean);
+            } else {
+                throw new Error('Signed integer required');
+            }
+        case ParameterType.String:
+            if (typeof userInput === 'string') {
+                const pSchema: StringType = paramSchema as StringType;
+                const bufferString: Buffer[] = [];
+                const length = userInput.length;
+                const stringLengthBuffer = serializeLength(
+                    length,
+                    pSchema.sizeLength
+                );
+                bufferString.push(stringLengthBuffer);
+                const stringBuffer = Buffer.from(userInput);
+                bufferString.push(stringBuffer);
+                return Buffer.concat(bufferString);
+            } else {
+                throw new Error('String required');
+            }
+        case ParameterType.Array:
+            return serializeArray(paramSchema as ArrayType, userInput);
+        case ParameterType.Struct:
+            return serializeStruct(paramSchema as StructType, userInput);
+        case ParameterType.AccountAddress:
+            const accountAddress = new AccountAddress(userInput);
+            return accountAddress.decodedAddress;
+        case ParameterType.Amount:
+            const GTUAmount = new GtuAmount(BigInt(userInput));
+            return encodeWord64(GTUAmount.microGtuAmount, true);
+        case ParameterType.Timestamp:
```

Comment:
Timestamp are actually expected to be datestrings:
https://developer.concordium.software/en/mainnet/smart-contracts/references/schema-json.html#timestamp